src/domain/app/header/languageDropdown/LanguageDropdown.tsx

```diff
@@ -6,26 +6,31 @@ import { useHistory, useLocation } from 'react-router-dom';
 import MenuDropdown, {
   MenuItem,
 } from '../../../../common/components/menuDropdown/MenuDropdown';
-import { SUPPORT_LANGUAGES } from '../../../../constants';
-import useLocale from '../../../../hooks/useLocale';
-import { Language } from '../../../../types';
+import { ROUTER_LANGUAGES, SUPPORT_LANGUAGES } from '../../../../constants';
+import { LanguageSelectorLanguage } from '../../../../types';
 import updateLocaleParam from '../../../../utils/updateLocaleParam';
 
 const LanguageDropdown: React.FC = () => {
-  const { t } = useTranslation();
+  const { t, i18n } = useTranslation();
   const history = useHistory();
   const { pathname, search } = useLocation();
-  const locale = useLocale();
+  const locale = i18n.language;
```

Comment:
Could useLocale just be updated to support 'cimode'?

Reply:
Nope, I don't think so, like I said in earlier comment: https://github.com/City-of-Helsinki/palvelutarjotin-admin/pull/177#issuecomment-834150214. When the 'cimode' is added there, the auto-suggest -widget, all kind of datepicker widgets, etc starts complaining about it, because they don't support 'cimode' as a language.

Reply:
yeah I see. That is problematic. Maybe history.push should have some abstraction that adds the locale behind the scenes and we don't have to mabually worry about it everhywhere. That would simplify things in this case too.